apollo-ios/Sources/Apollo/FieldSelectionCollector.swift

```diff
@@ -93,15 +93,26 @@ struct DefaultFieldSelectionCollector: FieldSelectionCollector {
                             info: info)
         }
 
-      case let .deferred(_, typeCase, _):
+      case let .deferred(condition, typeCase, _):
         // In Apollo's implementation (Router + Server) of deferSpec=20220824 ALL defer directives
         // will be honoured and sent as separate incremental responses. This means deferred
         // selection fields never need to be collected because they are parsed with the incremental
         // data, at which time they are no longer deferred.
         //
         // The deferred fragment identifiers still need to be collected becuase that is how the
         // user determines the state of the deferred fragment via the @Deferred property wrapper.
-        groupedFields.addDeferredFragment(typeCase)
+        var collectDeferredFragment: Bool = true
+        if let condition {
+          collectDeferredFragment = condition.evaluate(with: info.variables)
+        }
```

Comment:
```suggestion
        let isDeferred: Bool = {
          if let condition, !condition.evaluate(with: info.variables) {
            return false
          }
          return true
        }()
```

Reply:
I usually prefer a computed `let` to a `var` in situations like these (though that's not always possible).